### ISSUE TITLE
[wg/rdf-star] RDF-star Group Charter

### ISSUE THREAD

**Issue by pchampin** [w3c/strategy#467]
New charter proposal, reviewers please take note.

# Charter Review

Charter: https://w3c.github.io/rdf-star-wg-charter/

[diff from previous charter](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2F2022%2F08%2Frdf-star-wg-charter%2F&doc2=https%3A%2F%2Fw3c.github.io%2Frdf-star-wg-charter%2F)

 - [x] Existing WG recharter 

The group is chartered until August 2024, and is behind schedule on its deliverables. The group believes that it needs one more year to complete its work, so a regular charter extension (limited to 6 months) is not appropriate. Furthermore, the group would like to continue in maintenance mode once the deliverables are published (the published recommendations will [allow new features](https://www.w3.org/2023/Process-20231103/#allow-new-features)).

Except for the new schedule, and the transition to maintenance mode, the new charter does not contain any substansive change compared to the [previous one](https://github.com/w3c/strategy/issues/304), so we don't expect any major issue with horizontal reviews.

[issues](https://github.com/w3c/rdf-star-wg-charter/issues)

**Comment by himorin** [comment]
- why start date is the same as current running charter?
- success criteria has interoperability as `is expected to have at least two independent implementations of every feature defined in the specification.`, but why a word interoperable has removed?
- may not rely on wpt, but success criteria does not mention about open test suite
- performance HR (in coordination) is no longer exists

**Comment by pchampin** [comment]
> * why start date is the same as current running charter?

no particular reason; the WG considers this essentially as a  extension of its current charter -- the only reason to ask for a full-fledged rechartering is that we need more than 6 months. We can make the new charter start when the current one ends, if that's deemed more appropriate.

> *  success criteria has interoperability as `is expected to have at least two independent implementations of every feature defined in the specification.`, but why a word interoperable has removed?

The word "interoperable" is not in the current charter, I don't know exactly why. I suspect that it was not in the charter template when we created it –or it was removed by an unnoticed accident... I have no problem putting it back.

> * may not rely on wpt, but success criteria does not mention about open test suite

Same as above: I suspect this was not part of the charter template 2 years ago. No problem adding this, all the more that such an open test-suite already exists for RDF and SPARQL: https://github.com/w3c/rdf-tests

> * performance HR (in coordination) is no longer exists

?? it was never there, as far as I can tell... And I'm not sure it would be relevant for this WG.

**Comment by xfq** [comment]
> > * performance HR (in coordination) is no longer exists
> 
> ?? it was never there, as far as I can tell... And I'm not sure it would be relevant for this WG.

That has been [removed](https://github.com/w3c/charter-drafts/commit/68005355c92ec0e4cfe762d26f27f9c67ed7dc44) from the charter template, and should probably be removed from the [new charter](https://w3c.github.io/rdf-star-wg-charter/#coordination) as well.

**Comment by pchampin** [comment]
> > > * performance HR (in coordination) is no longer exists
> > 
> > 
> > ?? it was never there, as far as I can tell... And I'm not sure it would be relevant for this WG.
> 
> That has been [removed](https://github.com/w3c/charter-drafts/commit/68005355c92ec0e4cfe762d26f27f9c67ed7dc44) from the charter template, and should probably be removed from the [new charter](https://w3c.github.io/rdf-star-wg-charter/#coordination) as well.

Oh, sorry @himorin I misunderstood your comment about performance, and thanks @xfq for the clarification. +1 to remove it from the new charter.

**Comment by himorin** [comment]
https://www.w3.org/2022/07/immersive-web-wg-charter.html

**Comment by ruoxiran** [comment]
no comment or request from APA.

**Comment by simoneonofri** [comment]
Hi @pchampin:
 
I have seen that many deliverables are quite dated and perhaps earlier than was recommended Security and Privacy considerations:
 - [RDF 1.1 N-Quads](https://www.w3.org/TR/2014/REC-n-quads-20140225/): yes
 - [RDF 1.1 N-Triples](https://www.w3.org/TR/2014/REC-n-triples-20140225/): yes
 - [RDF 1.1 Semantics](https://www.w3.org/TR/2014/REC-rdf11-mt-20140225/): 
 - [RDF 1.1 TriG](https://www.w3.org/TR/2014/REC-trig-20140225/): yes
 - [RDF 1.1 Turtle](https://www.w3.org/TR/2014/REC-turtle-20140225/): yes
 - [RDF 1.1 XML Syntax](https://www.w3.org/TR/2014/REC-rdf-syntax-grammar-20140225/): no
 - [RDF Schema 1.1](https://www.w3.org/TR/2014/REC-rdf-schema-20140225/): no
 - [SPARQL 1.1 Overview](https://www.w3.org/TR/2013/REC-sparql11-overview-20130321/): no
 - [SPARQL 1.1 Query Language](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#security): yes 
 - [SPARQL 1.1 Update](https://www.w3.org/TR/2013/REC-sparql11-update-20130321/#security): yes
 - [SPARQL 1.1 Query Results JSON Format](https://www.w3.org/TR/2013/REC-sparql11-results-json-20130321/): no (but externally referenced)
 - [SPARQL Query Results XML Format (Second Edition)](https://www.w3.org/TR/2013/REC-rdf-sparql-XMLres-20130321/): no (but externally referenced)
 - [SPARQL 1.1 Service Description](https://www.w3.org/TR/2013/REC-sparql11-service-description-20130321/): no
  - [SPARQL 1.1 Update](https://www.w3.org/TR/2013/REC-sparql11-update-20130321/#security): yes

Mainly there are two elements (with examples from the specs):

**Formats**:

> N-Quads is a general-purpose assertion language; applications may evaluate given data to infer more assertions or to dereference IRIs, invoking the security considerations of the scheme for that IRI. Note in particular, the privacy issues in [RFC3023] section 10 for HTTP IRIs. Data obtained from an inaccurate or malicious data source may lead to inaccurate or misleading conclusions, as well as the dereferencing of unintended IRIs. Care must be taken to align the trust in consulted resources with the sensitivity of the intended use of the data; inferences of potential medical treatments would likely require different trust than inferences for trip planning.
N-Quads is used to express arbitrary application data; security considerations will vary by domain of use. Security tools and protocols applicable to text (e.g. PGP encryption, MD5 sum validation, password-protected compression) may also be used on N-Quads documents. Security/privacy protocols must be imposed which reflect the sensitivity of the embedded information.
N-Quads can express data which is presented to the user, for example, RDF Schema labels. Application rendering strings retrieved from untrusted N-Quads documents must ensure that malignant strings may not be used to mislead the reader. The security considerations in the media type registration for XML ([RFC3023] section 10) provide additional guidance around the expression of arbitrary data and markup.
N-Quads uses IRIs as term identifiers. Applications interpreting data expressed in N-Quads should address the security issues of Internationalized Resource Identifiers (IRIs) [RFC3987] Section 8, as well as Uniform Resource Identifier (URI): Generic Syntax [RFC3986] Section 7.
Multiple IRIs may have the same appearance. Characters in different scripts may look similar (a Cyrillic "о" may appear similar to a Latin "o"). A character followed by combining characters may have the same visual representation as another character (LATIN SMALL LETTER E followed by COMBINING ACUTE ACCENT has the same visual representation as LATIN SMALL LETTER E WITH ACUTE). Any person or application that is writing or interpreting data in Turtle must take care to use the IRI that matches the intended semantics, and avoid IRIs that make look similar. Further information about matching of similar characters can be found in Unicode Security Considerations [UNICODE-SECURITY] and Internationalized Resource Identifiers (IRIs) [RFC3987] Section 8.

**Query language**
> Exposing RDF data for update creates many security issues which all deployments must be aware of, and consider the risks involved. This submission discusses some of the potential issues. New security problems are discovered regularly, and each implementation introduces its own concerns. Consequently implementers should be aware that this is only a partial list containing possible issues, and cannot be considered complete nor authoritative.

> Write access to data makes it inherently vulnerable to malicious access. Standard access and authentication techniques should be used in any networked environment. In particular, HTTPS should be used, especially when implementing the SPARQL HTTP-based protocols. (i.e., encryption with challenge/response based password presentation, encrypted session tokens, etc). Some of the weak points addressed by HTTPS are: authentication, active session integrity between client and server, preventing replays, preventing continuation of defunct sessions.
SPARQL Update incurs all of the security concerns of SPARQL Query. In particular, stores which treat IRIs as dereferenceable need to protect against dereferenced IRIs from being used to invoke cross-site scripting attacks.
> Implementations will need to enforce their standard permissions scheme carefully. Permissions schemes always require careful design, and it is important to ensure that privileges in one area are not inadvertently applied to other parts of the system.
Systems that provide both read-only and writable interfaces can be subject to injection attacks in the read-only interface. In particular, a SPARQL endpoint with a Query service should be careful of injection attacks aimed at interacting with an Update service on the same SPARQL endpoint. Like any client code, interaction between the query service and the update service should ensure correct escaping of strings provided by the user.
> While SPARQL Update and SPARQL Query are separate languages, some implementations may choose to offer both at the same SPARQL endpoint. In this case, it is important to consider that an Update operation may be obscured to masquerade as a query. For instance, a string of unicode escapes in a PREFIX clause could be used to hide an Update Operation. Therefore, simple syntactic tests are inadequate to determine if a string describes a query or an update.

Thinking about specific Threats:
 - For Query languages: 
  -  Input: Injection/DoS attacks (e.g., SPARQL Injections)
  - Computing: Un-referencing IRI/elements, loading external sources (e.g. XML Entity Expansion style)
  - Output: XSS (or general reflection of uncontrolled text in responses, also with encoding)
 - For Formats: Encoding errors, although there may be problems related to serialization/unserialization of the graphs, similar to what happens with the [XML Signature Wrapping](https://www.nds.rub.de/media/nds/veroeffentlichungen/2012/08/22/BreakingSAML_3.pdf).
 
 And assumptions: 
 - Spoofing, Tampering: assuming SSL/TLS connection
 - Information Disclosure: assuming secure storage of data depending on their requirements
 
I also did a search on arxiv there are no papers describing additional vulnerabilities. On [SPARQL Injection](https://www.slideshare.net/simone.onofri/owasp-italy-day-sparql-injection-attacking-triple-store-semantic-web-application-security-sql-injection) a presentation from several years ago (in Italian).

If we think it can be useful to create a summary of potential threats, There are no further notes at present.

**Comment by plehegar** [comment]
No comments from PING

**Comment by pchampin** [comment]
@simoneonofri, thanks for the review. This will be valuable input for the specs, but I don't think it calls for any change on the charter itself. Would you agree?

**Comment by simoneonofri** [comment]
You're welcome, @pchampin; on a practical level, if the specs are updated, we can make a summary section.

**Comment by plehegar** [comment]
Various editorial updates in https://github.com/w3c/rdf-star-wg-charter/pull/60

**Comment by plehegar** [comment]
[Proposed W3C Charter: RDF-star Working Group (until 2024-10-03/04)](https://lists.w3.org/Archives/Public/public-new-work/2024Aug/0006.html)

**Comment by pchampin** [comment]
The vote is now closed, with a fair level of support for the new charter, and no formal objection.

Some changes on the text were suggested.

**Comment by pchampin** [comment]
> Some changes on the text were suggested.

https://github.com/w3c/rdf-star-wg-charter/pull/61

**Comment by pchampin** [comment]
> > Some changes on the text were suggested.
> 
> [w3c/rdf-star-wg-charter#61](https://github.com/w3c/rdf-star-wg-charter/pull/61)

**Comment by plehegar** [comment]
Proposed changes following AC review are under review until November 20.

**Comment by plehegar** [comment]
The proposed changes for the charter received a [formal objection](https://lists.w3.org/Archives/Public/public-review-comments/2024Nov/0003.html).

**Comment by plehegar** [comment]
[Decision of the W3C Council on the Formal Objection to the proposed RDF-star Working Group charter](https://www.w3.org/2025/04/rdf-star-council-report.html)

**Comment by plehegar** [comment]
[Announced](https://lists.w3.org/Archives/Member/w3c-ac-members/2025AprJun/0025.html)
